swarm/src/connection/pool.rs

```diff
@@ -692,17 +691,56 @@ where
                     let check_peer_id = || {
                         if let Some(peer) = expected_peer_id {
                             if peer != obtained_peer_id {
-                                return Err(PendingConnectionError::WrongPeerId {
-                                    obtained: obtained_peer_id,
-                                    endpoint: endpoint.clone(),
-                                });
+                                return match &endpoint {
+                                    ConnectedPoint::Dialer { address, .. } => {
+                                        Err(PoolEvent::PendingOutboundConnectionError {
+                                            id,
+                                            error: PendingOutboundConnectionError::WrongPeerId {
+                                                obtained: obtained_peer_id,
+                                                address: address.clone(),
+                                            },
+                                            peer: None,
+                                        })
```

Comment:
```suggestion
                                    ConnectedPoint::Dialer { address, .. } => {
                                        Err(PoolEvent::PendingOutboundConnectionError {
                                            id,
                                            error: PendingOutboundConnectionError::WrongPeerId {
                                                obtained: obtained_peer_id,
                                                address: address.clone(),
                                            },
                                            peer: Some(obtained_peer_id),
                                        })
```

Reply:
We needed to return `peer: expected_peer_id.or(Some(obtained_peer_id))`. 
That's what causing the tests to fail.

Reply:
But we already check above that `expected_peer_id` is either `None` or same as `obtained_peer_id`, so it shouldn't make a difference here. 
Am I missing something?

Reply:
Actually it was meant specifically for [this](https://github.com/akaladarshi/rust-libp2p/blob/7840463643e3de558d48eb5d6d254d4d1e6b9d84/swarm/src/connection/pool.rs#L702) because we are checking `expected_peer_id != obtained_peer_id `, but [here](
https://github.com/akaladarshi/rust-libp2p/blob/7840463643e3de558d48eb5d6d254d4d1e6b9d84/swarm/src/connection/pool.rs#L718) as you suggested it doesn't matter because we already have checked